app/src/androidTest/java/com/duckduckgo/app/notification/AndroidNotificationSchedulerTest.kt

```diff
@@ -433,12 +475,24 @@ class AndroidNotificationSchedulerTest {
         )
     }
 
-    private fun assertUnusedAppNotificationScheduled(workerName: String) {
-        assertTrue(getScheduledWorkers(NotificationScheduler.UNUSED_APP_WORK_REQUEST_TAG).any { it.tags.contains(workerName) })
+    private suspend fun givenStageIsUseOurAppNotification() {
+        whenever(privacyNotification.canShow()).thenReturn(false)
```

Comment:
do we need to mock this dependencies for this `given`?

Reply:
We do, yes because `scheduleNextNotification` also calls `scheduleInactiveUserNotifications` otherwise the test crashes because of a null.

Reply:
In that case, since it's not related with the `given` we are defining, I will recommend to create a new method called `givenNoInactiveUserNotifications` or similar and move those mock into that method.

Reply:
Done!